Types are broken from version 6 onwards

## Bug Report

Since updating to version `v6` and/ or `v7` I am getting the following error when using this library:

> Cannot find module 'deepmerge-ts' or its corresponding type declarations.
>   There are types at 'xxx/node_modules/deepmerge-ts/dist/node/index.d.mts', but this result could not be resolved under your current 'moduleResolution' setting. Consider updating to 'node16', 'nodenext', or 'bundler'.

It was working just fine in version `v5`.

### Expected behavior

The types should be working even if `moduleResolution` is set to `Node`.

### Steps to reproduce

Set up a project with the following typescript config:

```
 "module": "esnext",
  "moduleResolution": "Node",
```

When importing the library the error appears and the types are not working.

--- a/package.json
+++ b/package.json
@@ -41,6 +41,8 @@
     "import": "./dist/node/index.mjs",
     "require": "./dist/node/index.cjs"
   },
+  "main": "./dist/node/index.cjs",
+  "types": "./dist/node/index.d.cts",
   "files": [
     "dist/node/",
     "package.json",

